cmd/activator/main.go

```diff
@@ -85,27 +85,20 @@ var (
 	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 )
 
-func statReporter(statSink *websocket.ManagedConnection, stopCh <-chan struct{},
-	statChan <-chan []asmetrics.StatMessage, logger *zap.SugaredLogger) {
-	for {
-		select {
-		case sm := <-statChan:
-			go func() {
-				for _, msg := range sm {
-					b, err := json.Marshal(msg)
-					if err != nil {
-						logger.Errorw("Error while marshaling stat", zap.Error(err))
-					}
-					if err := statSink.SendRaw(gorillawebsocket.TextMessage, b); err != nil {
-						logger.Errorw("Error while sending stat", zap.Error(err))
-					}
+func statReporter(statSink *websocket.ManagedConnection, statChan <-chan []asmetrics.StatMessage,
+	logger *zap.SugaredLogger) {
+	for sm := range statChan {
+		go func(msgs []asmetrics.StatMessage) {
+			for _, msg := range msgs {
+				b, err := json.Marshal(msg)
+				if err != nil {
+					logger.Errorw("Error while marshaling stat", zap.Error(err))
 				}
-			}()
-		case <-stopCh:
-			// It's a sending connection, so no drainage required.
-			statSink.Shutdown()
-			return
-		}
+				if err := statSink.SendRaw(gorillawebsocket.TextMessage, b); err != nil {
```

Comment:
needs `else if` here. We don't want to try to send the message that failed serializing.

Reply:
Right, added a continue.